constantine/eth_verkle_ipa/ipa_prover.nim

```diff
@@ -40,62 +40,44 @@ import
 
 # Further reference refer to this https://dankradfeist.de/ethereum/2021/07/27/inner-product-arguments.html
 
-func genIPAConfig*(res: var IPASettings) : bool {.inline.} =
+func genIPAConfig*(res: var IPASettings) =
   # Initiates a new IPASettings
   # IPASettings has all the necessary information related to create an IPA proof
-  # such as SRS, precomputed weights for Barycentric formula
+  # such as crs, precomputed weights for Barycentric formula
 
   # The number of rounds for the prover and verifier must be in the IPA argument,
   # it should be log2 of the size of the input vectors for the IPA, since the vector size is halved on each round.
 
-  # genIPAConfig( ) generates the SRS, Q and the precomputed weights for barycentric formula. The SRS is generated
-  # as random points of the VerkleDomain where the relative discrete log is unknown between each generator.
-  var random_points: array[VerkleDomain, EC_P]
-  random_points.generate_random_points(uint64(VerkleDomain))
-  var q_val {.noInit.}: EC_P
-  q_val.fromAffine(Banderwagon.getGenerator())
-  var precomp {.noInit.}: PrecomputedWeights
-  precomp.newPrecomputedWeights()
-  var nr: uint32
-  nr.computeNumRounds(uint64(VerkleDomain))
-
-  res.numRounds = nr
-  res.Q_val = q_val
-  res.precompWeights = precomp
-  res.SRS = random_points
-
-  return true
-
-func populateCoefficientVector* (res: var openArray[Fr[Banderwagon]], ic: IPASettings, point: Fr[Banderwagon])=
-  var maxEvalPointInDomain {.noInit.}: Fr[Banderwagon]
-  maxEvalPointInDomain.fromInt(VerkleDomain - 1)
-
-  for i in 0 ..< res.len:
-    res[i].setZero()
-
-  if (maxEvalPointInDomain.toBig() < point.toBig()).bool == false:
-    let p = cast[int](point.toBig())
-    res[p].setOne()
-
-  else:
-    res.computeBarycentricCoefficients(ic.precompWeights, point)
-
-
-func coverIPARounds*(res: var IPAProof, transcript: var CryptoHash, ic: IPASettings, a: var openArray[Fr[Banderwagon]], b: var openArray[Fr[Banderwagon]], cb_c: var openArray[EC_P], q: EC_P, idx, rounds: var int): bool =
-
-  var a_view = a.toView()
-  var b_view = b.toView()
-  var cur_view = cb_c.toView()
-
-  var (a_L, a_R) = a_view.splitMiddle()
-  var (b_L, b_R) = b_view.splitMiddle()
-  var (G_L, G_R) = cur_view.splitMiddle()
+  # genIPAConfig( ) generates the crs, Q and the precomputed weights for barycentric formula. The crs is generated
+  # as random points of the EthVerkleDomain where the relative discrete log is unknown between each generator.
+  res.numRounds.computeNumRounds(uint64(EthVerkleDomain))
+  res.crs.generate_random_points()
+  res.domain.setupLinearEvaluationDomain()
+
+func coverIPARounds*(
+      res: var IPAProof,
+      transcript: var CryptoHash,
+      ic: IPASettings,
+      a: openArray[Fr[Banderwagon]],
+      b: openArray[Fr[Banderwagon]],
+      cb_c: openArray[ECP_TwEdwards_Aff[Fp[Banderwagon]]],
+      q: EC_P,
+      idx: var int,
+      rounds: int): bool =
```

Comment:
`rounds` is unused and already stored in IPASettings.

on line 127, the exit condition `if idx == 7` is hardcoded